app/src/utils/RedirectionUtils.js

```diff
@@ -461,16 +461,25 @@ export const redirectToFileMockEditorEditMock = (navigate, mockId) => {
   navigate(mockEditUrl);
 };
 
-export const redirectToMockEditorCreateMock = (navigate, newTab = false) => {
+export const redirectToMockEditorCreateMock = (navigate, newTab = false, collectionId = "") => {
+  const URL = collectionId
+    ? `${PATHS.MOCK_SERVER_V2.CREATE.ABSOLUTE}?collectionId=${collectionId}`
+    : PATHS.MOCK_SERVER_V2.CREATE.ABSOLUTE;
+
   if (newTab) {
-    window.open(PATHS.MOCK_SERVER_V2.CREATE.ABSOLUTE, "_blank");
+    window.open(URL, "_blank");
     return;
   }
-  navigate(PATHS.MOCK_SERVER_V2.CREATE.ABSOLUTE);
+
+  navigate(URL);
 };
 
-export const redirectToFileMockEditorCreateMock = (navigate, fileType) => {
-  const queryParam = fileType ? "file_type=" + fileType : "";
+export const redirectToFileMockEditorCreateMock = (navigate, fileType, collectionId = "") => {
+  let queryParam = "";
+
+  queryParam += fileType ? `file_type=${fileType}` : "";
+  queryParam += collectionId ? `&collectionId=${collectionId}` : "";
```

Comment:
if fileType is not present, then `collectionId` param will be added to the url with the `&` prefix. Its better to use `useSearchParam` here to embed multiple query params.

```js
  const queryParam = new URLSearchParams();
  if (fileType) iframeUrlParams.set("fileType", fileType);
  if (collectionId) iframeUrlParams.set("collectionId", collectionId);
```

```js
navigate(`${PATHS.FILE_SERVER_V2.CREATE.ABSOLUTE}?${queryParam.toString()}`);
``